From 4ed72fa658c03790700ba9084e9328fe38afdee9 Mon Sep 17 00:00:00 2001
From: Beniamino Galvani <bgalvani@redhat.com>
Date: Fri, 26 Apr 2019 10:49:21 +0200
Subject: [PATCH] device: fix reapply of MTU

When we set the MTU on the link we remember its previous source
(ip-config, parent-device or connection profile) and don't change it
again afterwards to avoid interfering with user's manual changes. The
only exceptions when we change it again are (1) if the parent device
MTU changes and (2) if the new MTU has higher priority than the one
previously set.

To allow a live reapply of the MTU property we also need to clear the
saved source, or the checks described above will prevent setting the
new value.

Fixes: 2f8917237fdf ('device: rework mtu priority handling')

https://bugzilla.redhat.com/show_bug.cgi?id=1702657
---
 src/devices/nm-device.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/devices/nm-device.c b/src/devices/nm-device.c
index 1a9a8a374..62d2e5783 100644
--- a/src/devices/nm-device.c
+++ b/src/devices/nm-device.c
@@ -11354,6 +11354,9 @@ check_and_reapply_connection (NMDevice *self,
 	s_ip6_old = nm_connection_get_setting_ip6_config (con_old);
 	s_ip6_new = nm_connection_get_setting_ip6_config (con_new);
 
+	/* Allow reapply of MTU */
+	priv->mtu_source = NM_DEVICE_MTU_SOURCE_NONE;
+
 	nm_device_reactivate_ip4_config (self, s_ip4_old, s_ip4_new);
 	nm_device_reactivate_ip6_config (self, s_ip6_old, s_ip6_new);
 
-- 
2.21.GIT

